src/Workspaces/Remote/ServiceHub/Services/NavigateToSearch/RemoteNavigateToSearchService.cs

```diff
@@ -47,75 +51,90 @@ public ValueTask HydrateAsync(Checksum solutionChecksum, CancellationToken cance
             string searchPattern,
             ImmutableArray<string> kinds,
-            RemoteServiceCallbackId callbackId,
             CancellationToken cancellationToken)
         {
-            return RunServiceAsync(solutionChecksum, async solution =>
+            return StreamWithSolutionAsync(solutionChecksum, SearchProjectWorkerAsync, cancellationToken).WithJsonRpcSettings(
+                new JsonRpcEnumerableSettings { MaxReadAhead = MaxReadAhead });
+
+            async IAsyncEnumerable<RoslynNavigateToItem> SearchProjectWorkerAsync(Solution solution, [EnumeratorCancellation] CancellationToken cancellationToken)
             {
                 var project = solution.GetRequiredProject(projectId);
-                var callback = GetCallback(callbackId, cancellationToken);
 
                 var priorityDocuments = priorityDocumentIds.SelectAsArray(d => solution.GetRequiredDocument(d));
 
-                await AbstractNavigateToSearchService.SearchProjectInCurrentProcessAsync(
-                    project, priorityDocuments, searchPattern, kinds.ToImmutableHashSet(), callback, cancellationToken).ConfigureAwait(false);
-            }, cancellationToken);
+                await foreach (var item in AbstractNavigateToSearchService.SearchProjectInCurrentProcessAsync(
+                    project, priorityDocuments, searchPattern, kinds.ToImmutableHashSet(), cancellationToken).ConfigureAwait(false))
+                {
+                    yield return item;
+                }
+            }
         }
 
-        public ValueTask SearchGeneratedDocumentsAsync(
+        public IAsyncEnumerable<RoslynNavigateToItem> SearchGeneratedDocumentsAsync(
             Checksum solutionChecksum,
             ProjectId projectId,
             string searchPattern,
             ImmutableArray<string> kinds,
-            RemoteServiceCallbackId callbackId,
             CancellationToken cancellationToken)
         {
-            return RunServiceAsync(solutionChecksum, async solution =>
+            return StreamWithSolutionAsync(solutionChecksum, SearchGeneratedDocumentsWorkerAsync, cancellationToken).WithJsonRpcSettings(
+                new JsonRpcEnumerableSettings { MaxReadAhead = MaxReadAhead });
+
+            async IAsyncEnumerable<RoslynNavigateToItem> SearchGeneratedDocumentsWorkerAsync(Solution solution, [EnumeratorCancellation] CancellationToken cancellationToken)
             {
                 var project = solution.GetRequiredProject(projectId);
-                var callback = GetCallback(callbackId, cancellationToken);
 
-                await AbstractNavigateToSearchService.SearchGeneratedDocumentsInCurrentProcessAsync(
-                    project, searchPattern, kinds.ToImmutableHashSet(), callback, cancellationToken).ConfigureAwait(false);
-            }, cancellationToken);
+                await foreach (var item in AbstractNavigateToSearchService.SearchGeneratedDocumentsInCurrentProcessAsync(
```

Comment:
Maybe dumb question here: why did this need an await + yield versus just returning directly the SearchGeneratedDocumentsInCurrentProcessAsync return? I don't see any other yielding/async but maybe I'm missing something.

Reply:
you are correct. t his doesn't need await/yield.  i can fixup later :)

Reply:
done in next pr.